src/components/time-input/TimeInput.tsx

```diff
@@ -105,6 +135,14 @@ const TimeInput = ({
         } = event;
         setDisplayTime(updatedValue);
         if (error) setError(undefined);
+        if (shouldAutoFormat) debouncedFormatDisplayTime(updatedValue);
```

Comment:
I think you can remove the option and do this unconditionally

Reply:
Yeah, I initially retained `shouldAutoFormat` to provide the option of formatting only on blur, but I think it's fine to run this function unconditionally